### ISSUE TITLE
License choice

### ISSUE THREAD

**Issue by Itarater** [Ellypse/LibMSP#8]
LibMSP is currently using Creative Commons Zero as a license, but I'd like to advocate for switching to a more conventional, albeit slightly more restrictive license. Namely, I'd like to advocate for switching to the [ISC License](https://choosealicense.com/licenses/isc/).

The effective differences between this license and CC0/public domain are the following:

* Copyright notices must be added for each author contributing under the license, and must be retained with all copies of the software (redistribution or modification, every copy).

* Permission notices (the full license text, as it's short) must be added to all copies of the software released under such a license, and must be retained by everyone who distributes or modifies it.

These are not particularly onerous, and would be accomplished by someone embedding the library just by not stripping the header out of the top of the file. But why would these be useful in any way over CC0?

* Clarity of license. Copyright laws have been unified between countries around the world, but public domain laws have not been. Some countries do not permit public domain dedications (continental Europe is often cited as being problematic), which causes works released under CC0 to fall back to the permissive copyright license anyway.

* Moral rights. Moral rights of authorship are not waived in some jurisdictions by a public domain dedication, but they are in others. Moral rights of authorship are not waived by copyright licenses in any jurisdictions, unless those licenses explicitly attempt to waive them (ISC does not).

* Plagiarism issues. Public domain dedications and the CC0 fallback license do not require any display of authorship on any works, anywhere, but this still can cause issues in academia, journalism, and elsewhere with plagiarism. Even if a work has waived the need for attribution, attribution is still necessary in some circumstances. Codifying that attribution into the software license makes it more reliable for people in the future to not find copies of the library with attribution records stripped from them.

How would this be done, if we decided to do so? To swap licenses from CC0/public domain to ISC, it's basically a case of needing to just swap out the license file and headers the next time a copyrightable modification is made. (My rewrite fork is already listed under ISC, though if there is not consensus on this, I'm fine with re-licensing my own work under CC0.)

This would apply the license to that modification and any future modifications only. Optionally, past authors may be contacted (namely Etarna, if anyone still has communication with her) for their assent in licensing their past work under ISC (effectively dual-licensing their original work). This is not necessary, but would legally allow us to put their names in a copyright statement in the header, as opposed to just noting their past authorship somewhere below the license header.

**Comment by Itarater** [comment]
If we decide to adopt the ISC license, the following is the specific text I prefer using:

```
Permission to use, copy, modify, and distribute this software for any
purpose with or without fee is hereby granted, provided that the above
copyright notice and this permission notice appear in all copies.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
```

Basically, this is the ISC license, as used by OpenBSD. It uses the word "and" in the first sentence, instead of "and/or". The "and/or" variant of the license is a bit more common these days, but there's some truly horrific ways you can abuse the phrase "and/or" in court, making me personally of the opinion that avoiding it is probably a good idea.

**Comment by Ellypse** [comment]
Like I said in our first conversation I would have preferred to keep the original licensing. That was my original intent in my own rewrite (as bad as it was code-wise). Etarna says in the licensing "It would be nice if changes to this were given away in the same manner, but you don't have to." so I thought it was indeed a nice gesture to continue his work on the same licensing.

Now I understand that the license you want to apply is less ambiguous and better for the project in the long run, and we should probably do it. But I wanted to give my honest opinion. Of course I have little to no knowledge on licensing matters, especially compared to you, and my view on this matter is purely from my feelings, and as such should probably be ignored.

Etarna replied to my private message on Curse about helping to maintain MyRolePlay in less than a week, so I'd say she would reply if she were contacted on the matter. Of course I understand that with the whole Curse/Twitch merge thing it's difficult for you to access Curse private messages right now. Maybe you could try sending an email to the address listed in the copyright note?

In conclusion, I would be okay to change the licensing for the good of the project if no other party disagrees with it.

**Comment by Itarater** [comment]
>Of course I have little to no knowledge on licensing matters, especially compared to you, and my view on this matter is purely from my feelings, and as such should probably be ignored.

I understand the inclination towards not changing things. It's unlikely to be a practical issue for a piece of software with such a small userbase. It's more of a philosophical concern, with the lack of a unifying standard of public domain, than something I expect to be a real issue.

The Open Source Initiative has [a couple FAQ questions](https://opensource.org/faq#public-domain) on the matter of both public domain and CC0, which are essentially answered with a long-worded shrug, as to whether releasing software as either public domain or CC0 is a good idea. In contrast, the Free Software Foundation gives it a [lukewarm thumbs up](https://www.gnu.org/licenses/license-list.en.html#CC0) (lukewarm being their maximum for anything that isn't strongly copyleft).

The main concerns are basically what I outlined -- the concept of public domain, as well as the ability to release a work into the public domain, are not unified world-wide, whereas the Berne Convention all-but-guarantees the same interpretation of copyright and copyright licenses worldwide. The ISC license itself is generally considered the most concise, permissive copyright license possible under the Berne Convention, insofar as granting nearly the same rights as public domain does, but doing it exclusively under copyright law.

So while I'm advocating for such a change on personal, philosophical grounds, it's not something that is likely to have any practical impact on the future of LibMSP in any way. (And I'm not a lawyer, either, I just spent an unhealthy amount of time obsessing over the state of copyright law in my youth. I was a really weird kid.)

I'll see about prodding Etarna to have her input on the issue, given her status as writing the original LibMSP code.

**Comment by Itarater** [comment]
I've been thinking, and I think keeping CC0 isn't really a problem for me, ultimately, as long as we add a new header to the source files.

Namely, this header should explicitly list attribution (as copyright headers do, except minus the copyright symbol), and explain that it's strongly recommended to retain the attribution text, as some jurisdictions may require so, even with public domain/CC0 works.

That would satisfy my concerns about the drawbacks of the license, while still retaining the license itself.

The header text could be something like this:
```
Author: Etarna Moonshyne
Author: Renaud "Ellypse" Parize
Author: Justin Snelgrove

This work is licensed under the Creative Commons Zero license. While not
required by the license, it is requested that you retain this notice and the
list of authors with any distributions, modified or unmodified, as it may be
required by law in some jurisdictions due to the moral rights of the authors.
```

Which would assuage my concerns about the lack of clarity surrounding public domain/CC0 and, especially, moral rights -- by basically informing users that they exist, and they might have to worry about them, but they probably don't if they just include the list of authors.

**Comment by Ellypse** [comment]
I just realized I never answered here. I like your proposition, and I like that you find a common ground. I'm really happy that we can keep the same licensing and continue the legacy of MSP in that way. Thanks ^^
